sklearn/svm/tests/test_svm.py

```diff
@@ -974,3 +974,21 @@ def test_ovr_decision_function():
     # Test if the first point has lower decision value on every quadrant
     # compared to the second point
     assert_true(np.all(pred_class_deci_val[:, 0] < pred_class_deci_val[:, 1]))
+
+def test_gamma_auto():
+    X, y = [[0.0], [1.0]], [0, 1]
+
+    msg = ("The default gamma parameter value 'auto', calculated as 1 / n_features,"
+        " is depreciated in version 0.19 and will be replaced by 'scale',"
+        " calculated as 1 / (n_features * X.std()) in version 0.21.")
+
+    assert_warns_message(DeprecationWarning,
+                         msg,
+                         svm.SVC(gamma='auto').fit, X, y)
```

Comment:
But this means that a user can't intentionally pass 'auto' without receiving a warning, which isn't great. We could solve this by making the default actually `'auto_deprecated'` which behaves like 'auto' with a warning. Using 'auto' explicitly would be same without warning.